airbyte-webapp/src/views/Connection/ConnectionForm/formConfig.tsx

```diff
@@ -197,55 +192,22 @@ function mapFormPropsToOperation(
   return newOperations;
 }
 
-function getDefaultCursorField(streamNode: SyncSchemaStream): string[] {
-  if (streamNode.stream.defaultCursorField.length) {
-    return streamNode.stream.defaultCursorField;
-  }
-  return streamNode.config.cursorField;
-}
-
-const useInitialSchema = (schema: SyncSchema): SyncSchema =>
-  useMemo<SyncSchema>(
-    () => ({
-      streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
-        const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
-
-        // If the value in supportedSyncModes is empty assume the only supported sync mode is FULL_REFRESH.
-        // Otherwise, it supports whatever sync modes are present.
-        const streamNode = nodeWithId.stream.supportedSyncModes?.length
-          ? nodeWithId
-          : setIn(nodeWithId, "stream.supportedSyncModes", [SyncMode.FullRefresh]);
-
-        // If syncMode isn't null - don't change item
-        if (streamNode.config.syncMode) {
-          return streamNode;
-        }
-
-        const updateStreamConfig = (config: Partial<AirbyteStreamConfiguration>): SyncSchemaStream => ({
-          ...streamNode,
-          config: { ...streamNode.config, ...config },
-        });
-
-        const supportedSyncModes = streamNode.stream.supportedSyncModes;
+export const calculateInitialCatalog = (
+  schema: SyncSchema,
+  supportedDestinationSyncModes: DestinationSyncMode[],
+  isEditMode?: boolean
+): SyncSchema => ({
+  streams: schema.streams.map<SyncSchemaStream>((apiNode, id) => {
+    const nodeWithId: SyncSchemaStream = { ...apiNode, id: id.toString() };
+    const nodeStream = verifySupportedSyncModes(nodeWithId);
```

Comment:
Agree, that also confused me.
As I understood(if we look at the comments I removed) sometimes we can get `supportedSyncModes: []`
That's what we need to check in edit and creation modes supportedSyncModes.
I don't know how we can get empty supportedSyncModes.